exp/controllers/azuremachinepool_controller.go

```diff
@@ -210,25 +211,57 @@ func (ampr *AzureMachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	logger = logger.WithValues("cluster", cluster.Name)
 
-	logger = logger.WithValues("AzureCluster", cluster.Spec.InfrastructureRef.Name)
-	azureClusterName := client.ObjectKey{
-		Namespace: azMachinePool.Namespace,
-		Name:      cluster.Spec.InfrastructureRef.Name,
-	}
-	azureCluster := &infrav1.AzureCluster{}
-	if err := ampr.Client.Get(ctx, azureClusterName, azureCluster); err != nil {
-		logger.V(2).Info("AzureCluster is not available yet")
-		return reconcile.Result{}, nil
-	}
+	var clusterScope azure.ClusterScoper
+	var groupScope groups.GroupScope
+	switch cluster.Spec.InfrastructureRef.Kind {
+	case "AzureCluster":
```

Comment:
I imagine we have this string hardcoded in enough places already that it's probably worth making a constant for (and also AzureManagedControlPlane).

(Maybe this would be a good first issue?)

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/4223